dbms/tests/queries/0_stateless/01018_empty_aggregation_filling.sql

```diff
@@ -0,0 +1,61 @@
+SELECT '--- Int Empty ---';
+
+SELECT arrayReduce('avgOrDefault', arrayPopBack([1]));
+SELECT arrayReduce('avgOrNull', arrayPopBack([1]));
+SELECT arrayReduce('stddevSampOrDefault', arrayPopBack([1]));
+SELECT arrayReduce('stddevSampOrNull', arrayPopBack([1]));
+SELECT arrayReduce('maxOrDefault', arrayPopBack([1]));
+SELECT arrayReduce('maxOrNull', arrayPopBack([1]));
+
+SELECT avgOrDefaultIf(x, x > 1) FROM (SELECT 1 AS x);
+SELECT avgOrNullIf(x, x > 1) FROM (SELECT 1 AS x);
+SELECT stddevSampOrDefaultIf(x, x > 1) FROM (SELECT 1 AS x);
+SELECT stddevSampOrNullIf(x, x > 1) FROM (SELECT 1 AS x);
+SELECT maxOrDefaultIf(x, x > 1) FROM (SELECT 1 AS x);
+SELECT maxOrNullIf(x, x > 1) FROM (SELECT 1 AS x);
+
+SELECT '--- Int Non-empty ---';
+
+SELECT arrayReduce('avgOrDefault', [1]);
+SELECT arrayReduce('avgOrNull', [1]);
+SELECT arrayReduce('stddevSampOrDefault', [1]);
+SELECT arrayReduce('stddevSampOrNull', [1]);
+SELECT arrayReduce('maxOrDefault', [1]);
+SELECT arrayReduce('maxOrNull', [1]);
+
+SELECT avgOrDefaultIf(x, x > 0) FROM (SELECT 1 AS x);
+SELECT avgOrNullIf(x, x > 0) FROM (SELECT 1 AS x);
+SELECT stddevSampOrDefaultIf(x, x > 0) FROM (SELECT 1 AS x);
+SELECT stddevSampOrNullIf(x, x > 0) FROM (SELECT 1 AS x);
+SELECT maxOrDefaultIf(x, x > 0) FROM (SELECT 1 AS x);
+SELECT maxOrNullIf(x, x > 0) FROM (SELECT 1 AS x);
+
+SELECT '--- Other Types Empty ---';
+
+SELECT arrayReduce('maxOrDefault', arrayPopBack(['hello']));
+SELECT arrayReduce('maxOrNull', arrayPopBack(['hello']));
+
+SELECT arrayReduce('maxOrDefault', arrayPopBack(arrayPopBack([toDateTime('2011-04-05 14:19:19'), null])));
```

Comment:
JFYI, this can be written more explicitly with cast, like `SELECT arrayReduce('maxOrDefault', cast([], 'Array(Nullable(DateTime))'));`
You don't have to change it, the current way is also OK.